packages/@uppy/transloadit/src/AssemblyOptions.ts

```diff
@@ -14,23 +16,41 @@ function validateParams (params) {
       params = JSON.parse(params)
     } catch (err) {
       // Tell the user that this is not an Uppy bug!
-      throw new ErrorWithCause('Transloadit: The `params` option is a malformed JSON string.', { cause: err })
+      throw new ErrorWithCause(
+        'Transloadit: The `params` option is a malformed JSON string.',
+        { cause: err },
+      )
     }
   }
 
-  if (!params.auth || !params.auth.key) {
-    throw new Error('Transloadit: The `params.auth.key` option is required. '
-      + 'You can find your Transloadit API key at https://transloadit.com/c/template-credentials')
+  if (!params!.auth || !params!.auth.key) {
+    throw new Error(
+      'Transloadit: The `params.auth.key` option is required. ' +
+        'You can find your Transloadit API key at https://transloadit.com/c/template-credentials',
+    )
   }
 }
+export type OptionsWithRestructedFields = Omit<Options, 'fields'> & {
+  fields: Record<string, string | number>
+}
 
 /**
  * Combine Assemblies with the same options into a single Assembly for all the
  * relevant files.
  */
-function dedupe (list) {
-  const dedupeMap = Object.create(null)
-  for (const { fileIDs, options } of list.filter(Boolean)) {
+function dedupe(
+  list: Array<
+    { fileIDs: string[]; options: OptionsWithRestructedFields } | undefined
```

Comment:
Maybe consider declaring it as a type so we can reuse it instead of duplicating it